quantization/existing-data.ipynb

```diff
@@ -0,0 +1,199 @@
+    "      input_type='search_document',\n",
+    "      embedding_types=[\"float\"] # Can also be \"int8\" or \"ubinary\" (int1)\n",
+    "    )\n",
+    "    embedding = response.embeddings.float[0]\n",
+    "    return embedding\n",
+    "\n",
+    "# Define function to convert embeddings to BSON-compatible format\n",
+    "def generate_bson_vector(vector, vector_dtype):\n",
+    "    return Binary.from_vector(vector, vector_dtype)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Connect to your Atlas cluster\n",
+    "mongo_client = pymongo.MongoClient(\"<ATLAS-CONNECTION-STRING>\")\n",
+    "db = mongo_client[\"sample_airbnb\"]\n",
+    "collection = db[\"listingsAndReviews\"]\n",
+    "\n",
+    "# Filter to exclude null or empty summary fields\n",
+    "filter = { \"summary\": {\"$nin\": [None, \"\"]} }\n",
+    "\n",
+    "# Get a subset of documents in the collection\n",
+    "documents = collection.find(filter).limit(50)\n",
+    "\n",
+    "# Initialize the count of updated documents\n",
+    "updated_doc_count = 0"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "for doc in documents:\n",
+    "    # Generate embeddings based on the summary\n",
+    "    summary = doc[\"summary\"]\n",
+    "    embedding = get_embedding(summary)  # Get float32 embedding\n",
+    "\n",
+    "    # Convert float32 embeddings into BSON format\n",
+    "    bson_vector = generate_bson_vector(embedding, BinaryVectorDtype.FLOAT32)\n",
+    "\n",
+    "    # Uncomment one of following lines if you specified a different data type in the get_embedding function\n",
+    "    # bson_vector = generate_bson_vector(embedding, BinaryVectorDtype.INT8)\n",
+    "    # bson_vector = generate_bson_vector(embedding, BinaryVectorDtype.PACKED_BIT) # int1\n",
+    "\n",
+    "    # Update the document with the BSON embedding\n",
+    "    collection.update_one(\n",
+    "        {\"_id\": doc[\"_id\"]},\n",
+    "        {\"$set\": {\"embedding\": bson_vector}}\n",
+    "    )\n",
+    "    updated_doc_count += 1\n",
+    "\n",
+    "print(f\"Updated {updated_doc_count} documents with BSON embeddings.\")"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from pymongo.operations import SearchIndexModel\n",
+    "\n",
+    "# Define and create the vector search index\n",
+    "vector_search_index_definition = {\n",
+    "  \"fields\":[\n",
+    "    {\n",
+    "      \"type\": \"vector\",\n",
+    "      \"path\": \"embedding\",\n",
+    "      \"similarity\": \"euclidean\",\n",
+    "      \"numDimensions\": 1024\n",
+    "    }\n",
+    "  ]\n",
+    "}\n",
+    "\n",
+    "search_index_model = SearchIndexModel(definition=vector_search_index_definition, name=\"<INDEX-NAME>\", type=\"vectorSearch\")\n",
```

Comment:
Same as below - this code doesn't include the polling section, so we should either _add_ the polling section or give the user some indication that they have to wait for the index to build before they can run the next block.